Lib/test/test_capi/test_long.py

```diff
@@ -80,13 +80,10 @@ def test_long_checkexact(self):
     def test_long_fromdouble(self):
         # Test PyLong_FromDouble()
         fromdouble = _testcapi.pylong_fromdouble
-        self.assertEqual(fromdouble(5.0), 5)
-        self.assertEqual(fromdouble(5.1), 5)
-        self.assertEqual(fromdouble(5.9), 5)
-        self.assertEqual(fromdouble(-5.1), -5)
-        self.assertEqual(fromdouble(-5.9), -5)
-        self.assertEqual(fromdouble(sys.float_info.max), int(sys.float_info.max))
-        self.assertEqual(fromdouble(-sys.float_info.max), -int(sys.float_info.max))
+        float_max = sys.float_info.max
+        for value in (5.0, 5.1, 5.9, -5.1, -5.9, 0.0, -0.0, float_max, -float_max):
+            with self.subTest(value=value):
+                self.assertEqual(fromdouble(value), int(value))
```

Comment:
This test is now suspicious. What if int() executes exactly the same code path, PyLong_FromDouble()?

I would prefer to hardcode the expected result:

```
for value, expected in (
    (5.0, 5),
    ...
    (float_max, int(float_max)),
):
    ...
```